### PR TITLE
Upgrade alacritty_terminal in hopes to avoid PTY poll failing

### PR DESCRIPTION
**Pull request by mrnugget**
We saw stack traces in our #panic channel pop up that failed on this line:

    https://github.com/zed-industries/alacritty/blob/33306142195b354ef3485ca2b1d8a85dfc6605ca/alacritty_terminal/src/event_loop.rs#L323-L324

With this message:

    thread 'PTY reader' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 9, kind: Uncategorized, message: "Bad file descriptor" }'
    /Users/administrator/.cargo/git/checkouts/alacritty-afea874b09a502a5/3330614/alacritty_terminal/src/event_loop.rs:324

We don't know how to reproduce the error. It doesn't seem related to the number of open PTY handles, because `openpty` itself didn't fail. We can only assume that something went wrong between `openpty` and the setup of the polling.

Since Alacritty itself changed its polling mechanism significantly by switching from `mio` to `polling` (https://github.com/alacritty/alacritty/pull/6846) we upgraded with the hope that this will fix the bug.

Release Notes:

- Upgraded alacritty_terminal to newest version in order to hopefully fix a rare panic that can occur when starting a new terminal.

### CODE REVIEW

**Review by mikayla-maki — DISMISSED** [review]
Looks like it's mostly import changes, lgtm!

**Comment by mikayla-maki** [comment]
Ahh, it seems I was a bit hasty. This PR seems to remove the ability to set environment variables via the settings file. I'll see if I can restore it :)

File: crates/terminal/src/terminal.rs [hunk 1]
```diff
@@ -312,43 +304,38 @@ impl TerminalBuilder {
         env.insert("LC_ALL".to_string(), "en_US.UTF-8".to_string());
         env.insert("ZED_TERM".to_string(), true.to_string());
 
-        let alac_scrolling = Scrolling::default();
-        // alac_scrolling.set_history((BACK_BUFFER_SIZE * 2) as u32);
-
         let config = Config {
-            pty_config: pty_config.clone(),
-            env,
```
Comment (mikayla-maki) [inline crates/terminal/src/terminal.rs:320]: This env variable needs to be used somewhere

**Review by mikayla-maki — APPROVED** [review]
lgtm :)

**Comment by mrnugget** [comment]
> Ahh, it seems I was a bit hasty. This PR seems to remove the ability to set environment variables via the settings file. I'll see if I can restore it :)

Ahh, woops, we forgot that!

**Pull request merged**